text/metrics/0156-counter-gauge-model.md

```diff
@@ -0,0 +1,356 @@
+
+For example, a Gauge for expressing a vehicle's speed relative to the
+ground can be expressed either as the speed of its midpoint or by an
+independent measurement of the speed of each wheel.
+
+```
+speedGauge.Set(vehicleSpeed)
+```
+
+This can be replaced by one Gauge per wheel, since wheel speed and
+vehicle speed each define vehicle speed relative to the ground:
+
+```
+for i := 0; i < 4; i++ {
+  speedGauge.Set(wheelSpeed[i], { 'wheel': i })
+}
+```
+
+This form of Gauge rewrite is generally useful to capture additional
+measurements by creating distinct metric streams.
+
+## Meaning-preserving attribute erasure
+
+Several rules for rewriting metric events that preserve meaning have
+been shown above, focused on introducing new attributes and new
+measurements in ways do not change existing meaning or alter existing
+interpretations.
+
+Removing attributes from metric events does not, by definition, change
+their meaning, since attributes are interpreted as event selectors.
+Removing attributes from aggregated streams of OpenTelemetry Metrics
+data requires attention to the meaning being conveyed.
+
+Safe attribute erasure for OpenTelemetry Metrics streams is specified
+in a way that preserves meaning while removing only the forms of
+interpretation that made use of the erased attribute.
+
+_Reaggregation_ describes the process of combining OpenTelemetry
+metric streams.  For reaggregation to preserve meaning, Sum points
+must be combined by adding the inputs and Gauge points must be
+combined by selecting the last or random value.
+
+Note that erasure of attributes is defined so that it reverses the
+effect of introducing new measurements, and meaning is preserved in
+both directions.  This explains the definition for default
+aggregations that should be applied when re-aggreation OpenTelemetry
+metrics streams.  Sum streams are re-aggregated to preserve the
+implied rate, while Gauge points are reggregated to preserve the
+implied distribution of individual values.
+
+## Conveying meaning to the user
+
+OpenTelemetry states a requirement separating the API from the
+implementation, and to do so we have defined the meaning of metrics
+API events.  To preserve meaning through stages of reaggregation, we
+have specified distinct default aggregation rules for Counter and
+Gauge streams.
+
+When attributes are used with Counter and Gauge instruments, every
+distinct combination of attribute values determines a separate
+OpenTelemetry metrics stream, and each stream conveys meaning
+independently.  Because meaning is independent from the attributes
+used, the user may wish to disregard some attributes when interpreting
+a stream of metrics, restricting their attention to a subset of
+attributes.
+
+In database systems, this process is refered to as a performing a
+"Group-By", where aggregation is used to combine streams within each
+distinct set of grouped attributes.  For the benefit of OpenTelemetry
```

Comment:
This kind of falls over for me in practice.  Database Group-By is a full query language where NORMALLY you need to provide your own Aggregation semantic for every column of data *or* the query will fail.

What we're suggesting here is instead of:
```
SELECT AVG(count) FROM x GROUP BY label
```

The default aggregation function is *infferred* by otel based on the metric type.

Two things:

1. We shouldn't prevent users from using different aggregation functions in practice.  Indeed, once they make it to PromQL (or MQL for GCP), they'll be able to do whatever they want here and need to track their own meaning.
2.I assume a lot of the gymnastics we go through around default aggregation is to avoid exposing a query-like langauge for "rewrite rule" style collection behavior. 

My $.02 here is the focus on giving users a way to solve "rewrite rules" use cases is good.  Making it as easy as possible is good.  If we can't explain what *is and isn't safe* in very simple terms, we might be in trouble.   If the "meaning" we retain isn't the one users wanted to retain, then we're not really adding value.